src/rust/engine/process_execution/src/remote.rs

```diff
@@ -229,26 +232,26 @@ impl super::CommandRunner for CommandRunner {
                         )).to_boxed()
                       } else {
                         // maybe the delay here should be the min of remaining time and the backoff period
-                        Delay::new(Duration::from_millis(backoff_period))
-                          .map_err(move |e| {
-                            format!(
-                              "Future-Delay errored at operation result polling for {}, {}: {}",
-                              operation_name, description, e
-                            )
-                          }).and_then(move |_| {
-                            future::done(
-                              operations_client
-                                .get_operation_opt(
-                                  &operation_request,
-                                  command_runner3.call_option(),
-                                ).or_else(move |err| {
-                                  rpcerror_recover_cancelled(operation_request.take_name(), err)
-                                }).map(OperationOrStatus::Operation)
-                                .map_err(rpcerror_to_string),
-                            ).map(move |operation| {
-                              future::Loop::Continue((history, operation, iter_num + 1))
-                            }).to_boxed()
+                        Delay::new_handle(
+                          Instant::now() + Duration::from_millis(backoff_period),
```

Comment:
Is this the essence of the change? Setting the timeout to be an absolute timestamp?

Reply:
The crucial thing here is passing the second argument, the handle to the thread on which the future will be scheduled. And that that thread is started and stopped by context.rs's `Core` when we fork.

Setting the timeout to be absolute is sadly only required because there's no `Delay::at_handle` which takes a `Duration` and a `TimerHandle`.

Reply:
Got it. Thanks@